### ISSUE TITLE
Larry said there's a way to speed this up

### ISSUE THREAD

**Issue by robbyblum** [internet-trains/ottd-util#1]
https://github.com/internet-trains/ottd-util/blob/97824aef4a63185626a101b993c5e2cccc781871/random.nut#L6

I will never let you forget anything

**Comment by larryfenn** [comment]
we don't need to fill the array up more than the result of GSRandRange. should cut memory usage (and probably run time) in half in the average case.

**Comment by robbyblum** [comment]
Iterating over the GSList entries has a deterministic order, right?

**Comment by larryfenn** [comment]
Yes, provided there is no other operation going on to GSList. There may be a race condition if elsewhere in the program GSList is being resorted.

**Comment by robbyblum** [comment]
Mmm, good to know. I imagine that would matter more for scripts that run during gameplay than for the industry constructor.

**Comment by larryfenn** [comment]
It should be stated that the current implementation is also potentially faulty to a race condition

**Comment by larryfenn** [comment]
While I have not sped this up I realized this was itself a way better implementation of random access than what I had in the industry placer.
